crates/services/txpool/src/service.rs

```diff
@@ -136,12 +138,29 @@ where
                 should_continue = false;
             }
             new_transaction = self.gossiped_tx_stream.next() => {
-                if let Some(GossipData { data: Some(tx), .. }) = new_transaction {
+                if let Some(GossipData { data: Some(_), .. }) = &new_transaction {
```

Comment:
What was the problem with the previous code `let Some(GossipData { data: Some(tx), .. }) = new_transaction`?

Seems it does the same in a strange way=D

Reply:
It's because we want to use `take_data` to move the tx data out of gossip without cloning. It might be better to just expose the peer id in the interfaces so we don't need to worry about passing the GossipData back and forth between p2p and services receiving messages.

Reply:
yeah it's cuz of ownership, if I didn't check it with a reference it would've been consumed and later unusable for reporting validation, having said that the interface does leave a lot to be desired so we should improve it.

Reply:
https://github.com/FuelLabs/fuel-core/pull/942